lib/src/components/NxToast/NxToastContainer.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import NxToast from './NxToast';
+import { NxToastContainerProps, nxToastContainerPropTypes } from './types';
+import classNames from 'classnames';
+
+const NxToastContainer = (props: NxToastContainerProps) => {
+  const { toasts } = props,
+      className = classNames('nx-toast__wrapper'),
+      toastContainerRef = useRef<HTMLDivElement | null>(null);
+
+  const [domReady, setDomReady] = React.useState(false);
+
+  useEffect(() => {
+    setDomReady(true);
+  }, []);
+
+  //Create a portal to .nx-page-main if exists, if not, default to document.body
+  const domNodeForPortal = document.querySelector('.nx-page-main') as HTMLElement || document.body;
+
+  if (!domReady) {
+    return null;
+  }
+
+  return createPortal(
+    <div className={className} ref={toastContainerRef}>
+      <div className="nx-toast__container">
+        {toasts.sort((a, b) => (a.toastId > b.toastId) ? -1 : 1).map(toast => (
```

Comment:
Why is the sorting necessary? Aren't they always already going to be in this order?

Reply:
We discovered that without the sort method, the toasts would be stacking in the reverse order that is intended (the oldest on the top, and the newest at the bottom). This sort helps reverse the order so the newest remains on top.

Reply:
> Why is the sorting necessary? Aren't they always already going to be in this order?

We discovered that without the sort method, the toasts would be stacking in the reverse order that is intended (the oldest on the top, and the newest at the bottom). This sort helps reverse the order so the newest remains on top.

Reply:
> We discovered that without the sort method, the toasts would be stacking in the reverse order that is intended (the oldest on the top, and the newest at the bottom). This sort helps reverse the order so the newest remains on top.

I think a better solution would be to construct the list in the correct order in the first place, e.g. by adjusting `addToast` in `NxToastProvider` to prepend instead of append.

Reply:
> > We discovered that without the sort method, the toasts would be stacking in the reverse order that is intended (the oldest on the top, and the newest at the bottom). This sort helps reverse the order so the newest remains on top.
> 
> I think a better solution would be to construct the list in the correct order in the first place, e.g. by adjusting `addToast` in `NxToastProvider` to prepend instead of append.

Readjusted addToast to prepend toasts/remove the sort method here : [9b89d80](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/9b89d80a212b8d790ec8426bcd0959151f944e64)